### ISSUE TITLE
[Doc]: Tk embedding example crashes Spyder

### ISSUE THREAD

**Issue by igurin-invn** [matplotlib/matplotlib#23334]
### Documentation Link

https://matplotlib.org/stable/gallery/user_interfaces/embedding_in_tk_sgskip.html

### Problem

Executing the code crashes Spyder

### Suggested improvement

Replace this:

```button_quit = tkinter.Button(master=root, text="Quit", command=root.quit)```

with this:

```button_quit = tkinter.Button(master=root, text="Quit", command=root.destroy)```

Thanks to [StackOverflow](https://stackoverflow.com/questions/31150795/why-quit-command-doesnt-work).

**Comment by tacaswell** [comment]
Does that still work as intended as a stand-alone script? 

What exactly do you mean by "crash" here?  My guess is that `root.quit` is someplace inside doing `sys.exit` or equivalent so the Python process is exiting as instructed.  Does this take down all of spyder or just the sub-shell running python?

**Comment by igurin-invn** [comment]
I copied the example into a script and ran it, leading to a crash. I then removed everything matplotlib-related to isolate the issue to this line. I'm using Spyder, but the same thing happens in a "bare" IPython prompt.

The behavior is that the Tk window hangs. The rest of Spyder is still alive enough to accept a [Ctrl]-[.] to restart the kernel, but it experiences more errors when restarting, and the only way out is to restart all of Spyder.

* Spyder version: 5.3.1 None
* Python version: 3.10.5 64-bit
* Qt version: 5.15.3
* PyQt5 version: 5.15.4
* Operating System: Windows 10

**Comment by daniilS** [comment]
@tacaswell I agree that `root.destroy` would be better here: `root.quit` just exits the Tcl/Tk mainloop without destroying any widgets. The window only closes in the standalone script because the script itself terminates. `quit` can be useful when you want to exit the mainloop to continue control flow, but it can cause issues when something else is keeping the mainloop running, like when running the script from IDLE/Spyder/REPL.

An example that clearly shows the difference between the two when used in a standalone script:
```python

import tkinter as tk
from tkinter import ttk
from time import sleep

root = tk.Tk()
button_quit = ttk.Button(text="Quit", command = root.quit)
button_destroy = ttk.Button(text="Destroy", command=root.destroy)
button_quit.pack()
button_destroy.pack()

root.mainloop()
sleep(5)  # Destroy will have closed the window, Quit will keep it up until the script ends
```

**Comment by tacaswell** [comment]
@igurin-invn Given your observationsand @daniilS 's explanation, making your proposed change to the examples seems correct.  Can you open a PR doing so?

**Comment by QuLogic** [comment]
Fixed by #23360.
